specification/resources/resource-manager/readme.md

```diff
@@ -452,6 +464,26 @@ directive:
     from: managedapplications.json
     where: $.definitions.GenericResource.properties
     reason: managedBy is a top level property
+  - from: deploymentScripts.json
+    suppress: TrackedResourceGetOperation 
+    where: $.definitions.AzureCliScript
+    reason: Tooling issue.
```

Comment:
Curious what the actual tool issue is

Reply:
@nschonni we asked about the linter errors to the swagger team and they directed us to suppress the errors. Below is the copy of the email thread.

________________________________________
From: Filiz Topatan <Filiz.Topatan@microsoft.com>
Sent: Thursday, September 19, 2019 6:47 AM
To: Ruoxuan Wang; Zhiqing Qiu; Marcin Jastrzebski; Lei Wang (AZURE)
Cc: Catherine Zhu; Marc Reyhner
Subject: RE: AutoRest Linter Errors on Deployment Scripts Swagger
 
+ Marc, Marcin, Lei
 
Thanks for looking into this Ruoxuan. I’ll add the suppression accordingly.
 
From: Ruoxuan Wang <Ruoxuan.Wang@microsoft.com> 
Sent: Thursday, September 19, 2019 1:14 AM
To: Filiz Topatan <Filiz.Topatan@microsoft.com>; Zhiqing Qiu <Zhiqing.Qiu@microsoft.com>
Cc: Catherine Zhu <yungez@microsoft.com>
Subject: RE: AutoRest Linter Errors on Deployment Scripts Swagger
 
Hi, @Filiz Topatan
 
After syncing up with Zhiqing offline. The errors you mentioned are actually false positive. You can add suppression section in `readme.md`.  Such as
 
``` yaml
directive:
  - suppress: TrackedResourceGetOperation
    reason: <reason here>
    where:
                - $.definitions.CliScript
  - suppress: TrackedResourcePatchOperation
reason: <reason here>
where:
                - $.definitions.PowerShellScript
```
 
After debug it. The reason is complex. The rule “TrackedResourceGetOperation” and “TrackedResourcePatchOperation” not only check whether existing “get” or “patch” operation but also check “response.schema.reference” whether equal with name of definition. In your case, The name of CLIScript and PowerShellScript is different from the “response.schema.reference”. So the lint tool report the errors.
 
Thanks,
Ruoxuan
From: Filiz Topatan <Filiz.Topatan@microsoft.com>
Sent: Thursday, September 19, 2019 10:46 AM
To: Ruoxuan Wang <Ruoxuan.Wang@microsoft.com>; Zhiqing Qiu <Zhiqing.Qiu@microsoft.com>
Cc: Catherine Zhu <yungez@microsoft.com>
Subject: RE: AutoRest Linter Errors on Deployment Scripts Swagger
 
I already talked to Ruoxuan about the below issue and he mentioned he’s new in the area and so wanted me to add Zhiqing Qui to the email to get some help on the issue.
 
From: Filiz Topatan 
Sent: Wednesday, September 18, 2019 7:29 PM
To: Ruoxuan Wang <Ruoxuan.Wang@microsoft.com>; Zhiqing Qiu <Zhiqing.Qiu@microsoft.com>
Cc: Catherine Zhu <yungez@microsoft.com>
Subject: AutoRest Linter Errors on Deployment Scripts Swagger
 
Hi,
 
We have a swagger for a new resource type called deploymentScript in Microsoft.Resources namespace (we haven't yet created a PR for this new swagger). We are getting AutoRest linter warnings and errors during the validation however we couldn't figure out the reason for these errors. 
 
We suspect that these warnings and errors are somehow related to us using the discriminator pattern where we derive CliScript and PowerShellScript objects from the base class DeploymentScripts. The errors are due to not having Patch or Get operations for these derived types. It's as if the tool is not seeing the polymorphic relationship between the base class and it's derived classes where both Path and Get (and other) operations are defined. It'd be great if you can provide some input as to why these errors are generated and if there is an error on our side or a tooling issue causing them.
 
There are a few known/expected errors, I'm disregarding those and only include related errors in the attached LinterErrors file. Please see other attached files for the swagger and request example
 
We also have an ARM review of the swagger early next week, if it’s possible to shed a light on this issue, it is appreciated as we might get questions on it during the review.
 
Let me know if you need more information on the errors or our design.
 
Thanks
Filiz

Reply:
Thanks! I just like kicking the tires on the tools, and like seeing where the limitations pop up